lib/install/engine/cli/cli.go

```diff
@@ -73,7 +74,15 @@ type Config struct {
 
 // Execute executes the installer steps.
 // Implements installer.Engine
-func (r *Engine) Execute(ctx context.Context, installer install.Interface, config install.Config) (err error) {
+func (r *Engine) Execute(ctx context.Context, installer install.Interface, config install.Config) (dispatcher.Status, error) {
+	err := r.execute(ctx, installer, config)
+	if err != nil {
+		return dispatcher.StatusUnknown, trace.Wrap(err)
+	}
+	return dispatcher.StatusCompleted, nil
```

Comment:
I also opted to return the operation status after running the task instead of sending the completion event via a dispatcher as a way to simplify delivery of that final completion event. It is easier to reason about and to implement (bypassing the dispatcher indirection).